analyzere/base_resources.py

```diff
@@ -148,6 +149,13 @@ def to_dict(self):
             d[k] = to_dict(v)
         return d
 
+    def __hash__(self):
+        # Certainly not the most efficient hash, since anything without an ID
+        # has the same hash value.
+        if hasattr(self, "id"):
+            int(uuid.UUID(self.id))
+        return 0
```

Comment:
Ok, I think this is fine. I'm clearly not familiar enough with hash functions and sounds like you've got it under control here.